File: programmers/Lv2/주식가격.py
def solution(prices):
    answer, stack = [0] * len(prices), []
    for i, p in enumerate(prices):
        while stack and p < prices[stack[-1]]:
            j = stack.pop()
            answer[j] = i - j
        stack.append(i)
        
    while stack:
        k = stack.pop()
        answer[k] = len(prices) - (1+k)
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];

        for (int i = 0; i < prices.length; i++) {
            int currentPrice = prices[i];
            int count = 0;
            for (int j = i + 1; j < prices.length; j++) {
                int nextPrice = prices[j];
                count++;

                if (nextPrice < currentPrice) {
                    break;
                }
            }
            answer[i] = count;
        }

        return answer;
    }
}